helpers/getFiles.js

```diff
@@ -0,0 +1,46 @@
+const fs = require('fs');
+const path = require('path');
+
+const preCode = `
+const { useState, useEffect } = React;
+`;
+
+const postCode = (name) => `
+render(<${name} />);
+`;
+
+function getFiles(window) {
+    let editor = window.activeTextEditor;
+    if (!editor) return;
+
+    const doc = editor.document;
+    const fileName = doc.fileName;
+    const parentDir = path.dirname(fileName);
+
+    const componentSpecUrl = `${parentDir}/component-engine.json`;
+
+    let componentSpec;
+    if (componentSpecUrl === fileName) {
+        componentSpec = JSON.parse(doc.getText());
```

Comment:
Need some way to handle if JSON parse fails, one way is to start just accepting that the whole `getFiles` can throw an error. But then I would double down on that.
ie 
```js
 let editor = window.activeTextEditor;
	    if (!editor) throw new Error('No active text editor');
```

```js
    if (!componentSpec) throw new Error('Cannot find component Spec');
```

```js
    if (!component) throw new Error('Cannot find component');
```


Then when you consume it:

```js
try {
  const dataObject = getFiles(window);
  panel.webview.postMessage(dataObject);
}  catch(error) {
  panel.webview.postMessage({ error });
} 
```

I think in general we should go for the throw and catch model, when ever you do a null check, throw if it's not right. Let's get as much info surfaced to the user rather than gracefully failing.

Reply:
Yeah this is 100% right. I have a note next to my laptop reminding me to fix this and i forgot anyway haha

Reply:
As soon as i changed it to sync rather than callback I lost the initial error handling i had written. Ill update this